app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php

```diff
@@ -0,0 +1,138 @@
+
+class CreateLifeEvent extends BaseService implements ServiceInterface
+{
+    private LifeEvent $lifeEvent;
+
+    private Collection $partipantsCollection;
+
+    private array $data;
+
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules(): array
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'vault_id' => 'required|integer|exists:vaults,id',
+            'author_id' => 'required|integer|exists:users,id',
+            'life_event_type_id' => 'required|integer|exists:life_event_types,id',
+            'summary' => 'nullable|string|max:255',
+            'description' => 'nullable|string|max:65535',
+            'happened_at' => 'date|date_format:Y-m-d',
+            'costs' => 'nullable|integer',
+            'currency_id' => 'nullable|integer|exists:currencies,id',
+            'paid_by_contact_id' => 'nullable|integer|exists:contacts,id',
+            'duration_in_minutes' => 'nullable|integer',
+            'distance_in_km' => 'nullable|integer',
+            'from_place' => 'nullable|string|max:255',
+            'to_place' => 'nullable|string|max:255',
+            'place' => 'nullable|string|max:255',
+            'participant_ids' => 'required|array',
+        ];
+    }
+
+    /**
+     * Get the permissions that apply to the user calling the service.
+     *
+     * @return array
+     */
+    public function permissions(): array
+    {
+        return [
+            'author_must_belong_to_account',
+            'vault_must_belong_to_account',
+            'author_must_be_vault_editor',
+        ];
+    }
+
+    /**
+     * Create a life event.
+     *
+     * @param  array  $data
+     * @return LifeEvent
+     */
+    public function execute(array $data): LifeEvent
+    {
+        $this->data = $data;
+        $this->validate();
+        $this->store();
+        $this->associateParticipants();
+        $this->updateLastEditedDate();
+
+        return $this->lifeEvent;
+    }
+
+    private function validate(): void
+    {
+        $this->validateRules($this->data);
+
+        $lifeEventType = LifeEventType::findOrFail($this->data['life_event_type_id']);
+
+        $this->vault->lifeEventCategories()
+            ->findOrFail($lifeEventType->lifeEventCategory->id);
+
+        if (! is_null($this->data['paid_by_contact_id'])) {
+            $this->vault->contacts()
+                ->findOrFail($this->data['paid_by_contact_id']);
+        }
+
+        if (! is_null($this->data['currency_id'])) {
+            $this->account()->currencies()
+                ->findOrFail($this->data['currency_id']);
+        }
+
+        $this->partipantsCollection = collect();
+        foreach ($this->data['participant_ids'] as $participantId) {
+            $this->partipantsCollection->push(
+                $this->vault->contacts()->findOrFail($participantId)
+            );
+        }
```

Comment:
What do you think of:
```suggestion
        $this->partipantsCollection = collect($this->data['participant_ids'])
            ->map(fn(int $participantId): Contact => $this->vault->contacts()->findOrFail($participantId));
```

Note: it should be wise to check properly that `participant_ids` contains only `int` values, no?

Reply:
It works, dirty bastard.

Reply:
Changed